### PR TITLE
Extended eta v1

### PR DESCRIPTION
**Pull request by archiron**
#### PR description:

add some modifications in order to have histo with extended eta range from [-2.5, 2.5] to [-3.0, 3.0] to see improvement of electrons reco efficiency in the high eta region for Run3..
Modifications are made on the following files :
Validation/RecoEgamma/plugins/ElectronMcSignalPostValidator.cc Validation/RecoEgamma/plugins/ElectronMcSignalValidator.cc 
Validation/RecoEgamma/plugins/ElectronMcSignalValidator.h 

Validation/RecoEgamma/python/ElectronMcSignalValidator_gedGsfElectrons_cfi.py

Some other small (they do not concern the extended eta range) modifications are made on :
Validation/RecoEgamma/test/ElectronMcSignalValidation_gedGsfElectrons_cfg.py Validation/RecoEgamma/test/ElectronMcSignalPostValidation_cfg.py Validation/RecoEgamma/test/electronValidationCheck_Env.py 

This can be viewed with the following pictures:
h_mc_Eta_matched
actual eta_matched picture

h_mc_Eta_Extended_matched
same picture with extended eta range
h_ele_EoPVsEta_pfx
actual EoPVsEta_pfx picture
h_ele_EoPVsEtaExtended_pfx
same with extended eta range

The diagonal text is only here because there is no previous reference for the "extended range" histos.

#### PR validation:

compilation is OK, basics tests (scram b runtests or local tests) are OK too.
runTheMatrix.py -l limited -i all --ibeos tests are fine
@beaudett @swagata87

### CODE REVIEW

**Comment by swagata87** [comment]
type urgent 
 - currently eta-extended electrons are not monitored as part of release validation of electrons. But it would be necessary to monitor it, in particular in 12_4_0_pre4 and later, because egamma is putting in improved energy correction (and PF ID) for eta-extended electrons. So if possible, if this PR could go in pre4, it might be very useful.

**Comment by archiron** [comment]
> -code-checks
> 
> Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-37909/29938
> 
> * This PR adds an extra 48KB to repository
> 
> Code check has found code style and quality issues which could be resolved by applying following patch(s)
> 
> * **code-format**:
>   https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-37909/29938/code-format.patch
>   e.g. `curl -k https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-37909/29938/code-format.patch | patch -p1`
>   You can also run `scram build code-format` to apply code format directly

Please note that I use the code-format command before the pull request WITHOUT anything to change.

File: Validation/RecoEgamma/plugins/ElectronMcSignalValidator.cc [hunk 1]
```diff
@@ -3526,7 +3548,15 @@ void ElectronMcSignalValidator::analyze(const edm::Event &iEvent, const edm::Eve
     //------------------------------------
     passMiniAODSelection = bestGsfElectron.pt() >= 5.;
     isEBflag = bestGsfElectron.isEB();
-    isEEflag = bestGsfElectron.isEE();
+    isEEflag = bestGsfElectron.isEE() && (std::abs(mcIter->eta()) < 2.5);
```
Comment (perrotta) [inline Validation/RecoEgamma/plugins/ElectronMcSignalValidator.cc:3551]:
```suggestion
    isEEflag = bestGsfElectron.isEE() && (std::abs(mcIter->eta()) < maxAbsEta_);
```

File: Validation/RecoEgamma/plugins/ElectronMcSignalValidator.cc [hunk 2]
```diff
@@ -3556,6 +3586,7 @@ void ElectronMcSignalValidator::analyze(const edm::Event &iEvent, const edm::Eve
     h1_mc_Phi_matched->Fill(mcIter->phi());
     h1_mc_AbsEta_matched->Fill(std::abs(mcIter->eta()));
     h1_mc_Eta_matched->Fill(mcIter->eta());
+    h1_mc_Eta_Extended_matched->Fill(mcIter->eta());
```
Comment (perrotta) [inline Validation/RecoEgamma/plugins/ElectronMcSignalValidator.cc:3589]: You are duplicating the previous histo, only with a slightly wider x-range: is it really what you want?
One single histo with the widest range cannot suit as well?

File: Validation/RecoEgamma/plugins/ElectronMcSignalValidator.cc [hunk 3]
```diff
@@ -3805,6 +3836,7 @@ void ElectronMcSignalValidator::analyze(const edm::Event &iEvent, const edm::Eve
     // match distributions
     h1_ele_EoP->Fill(bestGsfElectron.eSuperClusterOverP());
     h2_ele_EoPVsEta->Fill(bestGsfElectron.eta(), bestGsfElectron.eSuperClusterOverP());
+    h2_ele_EoPVsEtaExtended->Fill(bestGsfElectron.eta(), bestGsfElectron.eSuperClusterOverP());
```
Comment (perrotta) [inline Validation/RecoEgamma/plugins/ElectronMcSignalValidator.cc:3839]: You are duplicating the previous histo, only with a slightly wider x-range: is it really what you want?
One single histo with the widest range cannot suit as well?

File: Validation/RecoEgamma/plugins/ElectronMcSignalValidator.cc (hunk 2 above)
Comment (beaudett) [reply Validation/RecoEgamma/plugins/ElectronMcSignalValidator.cc:3589]: Hi Andrea,
yes, it is intended. The idea is to introduce a new category of plots to monitor electrons in 2.5<|eta|<3 without affecting the rest of the plots. Florian

File: Validation/RecoEgamma/plugins/ElectronMcSignalValidator.cc (hunk 3 above)
Comment (beaudett) [reply Validation/RecoEgamma/plugins/ElectronMcSignalValidator.cc:3839]: Hi Andrea,
yes, it is intended. The idea is to introduce a new category of plots to monitor electrons in 2.5<|eta|<3 without affecting the rest of the plots. Florian

**Comment by emanueleusai** [comment]
+1

File: Validation/RecoEgamma/plugins/ElectronMcSignalValidator.cc (hunk 2 above)
Comment (perrotta) [reply Validation/RecoEgamma/plugins/ElectronMcSignalValidator.cc:3589]: > Hi Andrea, yes, it is intended. The idea is to introduce a new category of plots to monitor electrons in 2.5<|eta|<3 without affecting the rest of the plots. Florian

Well, the eta distributions are exactly the same in the two pair of plots that I pointed out, with only the x-axis range of the two histos being extended. I will not insist, of course: they are just two fully redundant validation histos which are added over a much larger number. Still, it doesn't seem really justified to me.

File: Validation/RecoEgamma/plugins/ElectronMcSignalValidator.cc (hunk 3 above)
Comment (beaudett) [reply Validation/RecoEgamma/plugins/ElectronMcSignalValidator.cc:3839]: Hi Andrea,
Thanks for your understanding. When we will add additional histos to monitor the electrons with |eta|>2.5, we will get rid of these two histograms. Florian

**Comment by perrotta** [comment]
+1

**Pull request merged**